### ISSUE TITLE
Add layers options to events map

### ISSUE THREAD

**Issue by andrewtavis** [activist-org/activist#718]
### Terms

- [X] I have searched [open and closed feature requests](https://github.com/activist-org/activist/issues?q=is%3Aissue+label%3Afeature)
- [X] I agree to follow activist's [Code of Conduct](https://github.com/activist-org/activist/blob/main/.github/CODE_OF_CONDUCT.md)

### Description

Based on discussions in the most recent dev sync, one final element that should be added to the map for events is a menu option to control certain layers. We would like this to have three options:

- One to toggle public transit maps
- One to toggle biking routes
- One that's a radio option for how the directions should be routes (walking, biking or driving)
  - This is related to #685 

Note that this is blocked by #673 that is where we're learning as a community how to make a clickable map option that calls a function to clear the directions.

### Contribution

Happy to work on this myself or support someone with interest when this is unblocked!

**Comment by andrewtavis** [comment]
As of now this issue would just focus on the bike route and transit map layers, and we could make another issue for switching the routing options once #685 is closed. I've just updated the [Figma designs for this](https://www.figma.com/file/I9McFfaLu1RiiWp5IP3YjE/activist_public_designs?type=design&node-id=9492%3A10277&mode=design&t=iAcJvM6U2RZjd60X-1), which can further be seen below:

![Mode=Light, Mobile=False, Type=Map, Disabled=False](https://github.com/activist-org/activist/assets/24387426/118240e0-fc1d-46e9-b7d8-d8c4916871a9)

![Mode=Light, Mobile=False, Type=Map, Disabled=False (2)](https://github.com/activist-org/activist/assets/24387426/e01399dc-909f-4cec-85bc-e5df92c463ca)

In the second image above we have the biking option as being selected (not reflected in the map). The selected option would change color, as would the layers color to indicate that one layer has been added.

CC @nicholas-vecchio, in case you'd have interest :)

**Comment by andrewtavis** [comment]
First step would be researching a way to add in the layers for the transit maps or biking maps onto the map, but hopefully there are extensions/plugins for this from OSM or another provider 😊

**Comment by nicholas-vecchio** [comment]
Thanks for the mention @andrewtavis I'd be happy to look into this :)

**Comment by andrewtavis** [comment]
Nice, @nicholas-vecchio! Really excited to hear what comes from this! Let me know if you need some support here, and happy to do a call to discuss at some point 😊

**Comment by andrewtavis** [comment]
Also wanted to mention that it'd be great to have you in the bi-weekly Saturday dev syncs, if you can make it ever :) Next one is [Saturday the 20th of April at 15:00 UTC](https://zonestamp.toolforge.org/1713625200), with the link and further details being in the [Development channel on Matrix](https://matrix.to/#/!CRgLpGeOBNwxYCtqmK:matrix.org?via=matrix.org&via=utwente.io&via=matrix.pierre-couy.fr).

**Comment by andrewtavis** [comment]
Note on this, @nicholas-vecchio, I've also just made #832 that's looking into the switch directions API/profile option, as we have been able to figure out how to do that at this point. Let me know if there's anything I can do here to support!

**Comment by andrewtavis** [comment]
Hey @nicholas-vecchio 👋 Quick check in to see how you're doing on this :) Let me know if you'd like to do a call at some point to check in on this! Would be happy to 😊

**Comment by nicholas-vecchio** [comment]
Hello @andrewtavis apologies for taking so long to get back to you! I'll begin experimenting to see if I can get cycling layers to work via an open source cycling map, but frankly I'm very unsure where it would be possible to source public transport tiles from.

**Comment by andrewtavis** [comment]
No stress, @nicholas-vecchio! I've had lots to do myself :) Looking at the public transit options for a quick moment, at first glance maybe [Transitland vector tiles](https://www.transit.land/documentation/vector-tiles) would work? I'd be happy to make an API key and send it along to you to play around with, which is how we did the captcha work with someone :) 

Let's focus on the bike routes first though! Let me know if there's anything I can do to help :)

**Comment by andrewtavis** [comment]
Hey @nicholas-vecchio 👋 We discussed this in the last dev sync :) What do you think about switching the context for this to being just for a boolean toggle for the bike routes? This would follow a lot of the work that @GrandSuccess87 did for the route profile switcher in #832, and would make use of the raster layers that OpenStreetMap has already :) We'd just switch the layer profile with something like the following:

```
https://b.tile-cyclosm.openstreetmap.fr/cyclosm/{z}/{x}/{y}.png
```

From there we'd expand to include the transit maps once @wkyoshida has checked out [Transitland](https://www.transit.land/) :)

Let me know how this sounds!

**Comment by andrewtavis** [comment]
With #949 merged in the only thing that we'd like from this at this point is for an option to show the public transit map, with this option being a boolean. FYI @nicholas-vecchio, @wkyoshida implemented the bike routes such that they're visible when the user is leveraging the bike directions option 🚲 @wkyoshida, what are your thoughts on the transit maps?

Will change the issue name to reflect the new scope :)

**Comment by andrewtavis** [comment]
Hey @nicholas-vecchio 👋 Checking in to see if you still have interest in working on this :) Please let us know if you need any support here! There's been lots of progress on the map component recently, so I'd definitely suggest pulling, and it'd be great if we could work on the transit maps or maybe finalize the routing options to add driving directions as an option (we'd do a different issue for this).

Hope you're well!

**Comment by andrewtavis** [comment]
Unassigning for now, @nicholas-vecchio, but please let us know if you'd like to pick up an issue in the future!

Hope all's well :)

**Comment by IWang20** [comment]
Hi, can I be assigned this issue?

**Comment by andrewtavis** [comment]
Sure thing, @IWang20! This'd be great to get some support on :) Maybe we could start with the research above and make a plan for where to go from there? Let us know what your thoughts are here 😊

**Comment by IWang20** [comment]
![Image](https://github.com/user-attachments/assets/9b28cb2f-f253-4833-a52d-2ca94aa6ca2a)
I see walking but I don't see the toggle for biking. Is that a change that hasn't been accepted into main yet?

**Comment by andrewtavis** [comment]
If you click the walking icon it will switch to biking :) What this issue is looking into is a third option that would be beneath the walking that would toggle the transit layer 🚂

**Comment by IWang20** [comment]
### Transit Fix Plan (in MediaMap.vue)

Add new variable definitions for transit: ProfileMap, new icon path, interface, ProfileOptions, and ProfileControl. 

I'm assuming the toggleLayerHandler will be left alone because it is just for walking? 

Rewrite the existing routeProfileHandler and setSelectedRoute for biking/walking to check if the current profile/route is transit. 

Add routeProfileHandler and setSelectedRoute for transit, which just checks if the current profile/route is biking/walking and sets current profile/route to transit. 

Add another button for transit using `map.addControl` to resetRouteProfileControl() that pretty much mirrors the walking/biking button, just with the appropriate functions.

**Comment by andrewtavis** [comment]
The above sounds good, @IWang20, but one thing I want to check on is that the plan is just to add a layer for the transit system and not to include a routing option that uses the public transit. If this can be added at some point, then really great, but let's get to it later :) Basically we just want another option beneath the walking/biking route option that then switches the layer for the transit system on or off 🚂

**Comment by IWang20** [comment]
Hey, is Transitland still the place to get tiles for transit routes?

**Comment by andrewtavis** [comment]
I would assume, so, but then if you found some alternatives I'd be happy to discuss them!
